### PR TITLE
Require break and continue labels inside static foreach.

### PR DESCRIPTION
**Pull request by tgehr**
According to http://forum.dlang.org/post/okjekv$2dqs$1@digitalmars.com .

### CODE REVIEW

File: test/compilable/staticforeach.d [hunk 1]
```diff
@@ -638,3 +638,36 @@ auto bug17660(){
...
+            }
+        }
+    })));
+    static assert(!is(typeof({
+        for(;;){
+            static foreach(i;0..1){
+                continue;
+            }
+        }
+    })));
```
Comment (jacob-carlborg) [inline test/compilable/staticforeach.d]: Tests that should fail to compile are usually placed inside `fail_compilation`, unless new conventions are used now.

Comment (tgehr) [inline test/compilable/staticforeach.d]: My test does not fail to compile. :P

Comment (WalterBright) [inline test/compilable/staticforeach.d]: Putting them in `fail_compilation/` allows for the exact error message generated to be tested, and so makes for a better test. There are plenty of examples in `fail_compilation/` to show how this is done - it's pretty easy.

Comment (tgehr) [inline test/compilable/staticforeach.d]: I have added fail_compilation tests.

**Review by andralex — APPROVED** [review]

**Pull request merged**